### PR TITLE
dummyhttp: init at 1.0.3

### PR DESCRIPTION
**Pull request by GuillaumeDesforges**
## Description of changes

Add dummyhttp to nixpkgs

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/du/dummyhttp/package.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    hash = "sha256-MDkyJAVNpCaAomAEweYrQeQWtil8bYo34ko9rAu+VBU=";
+  };
+
+  cargoHash = "sha256-JkA0qW/MQH+XmiD9eiT0s70HxNNYyk9ecBo4k5nUF10=";
+
+  meta = with lib; {
+    description = "Super simple HTTP server that replies a fixed body with a fixed response code";
+    homepage = "https://github.com/svenstaro/dummyhttp";
+    license = with licenses; [ mit ];
+    maintainers = with maintainers; [ ];
```
Comment (momeemt) [inline pkgs/by-name/du/dummyhttp/package.nix:24]:
```suggestion
    maintainers = with maintainers; [ GuillaumeDesforges ];
```

Could you add yourself to maintainers?

File: pkgs/by-name/du/dummyhttp/package.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+, darwin
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "dummyhttp";
+  version = "1.0.3";
+
+  src = fetchFromGitHub {
+    owner = "svenstaro";
+    repo = pname;
```
Comment (momeemt) [inline pkgs/by-name/du/dummyhttp/package.nix:13]: It is not recommended to reference `pname` from behind. ref: https://github.com/NixOS/nixpkgs/issues/277994

File: pkgs/by-name/du/dummyhttp/package.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    rev = "v${version}";
+    hash = "sha256-MDkyJAVNpCaAomAEweYrQeQWtil8bYo34ko9rAu+VBU=";
+  };
+
```
Comment (momeemt) [inline pkgs/by-name/du/dummyhttp/package.nix:17]: I suggest adding `darwin.apple_sdk.frameworks.Security` to buildInputs to build it in *-darwin. It failed to build without Security in aarch64-darwin, but I added the following then it succeeded.

```suggestion

  buildInputs = lib.optionals stdenv.isDarwin [
    darwin.apple_sdk.frameworks.Security
  ];

```

File: pkgs/by-name/du/dummyhttp/package.nix [hunk 4]
```diff
@@ -0,0 +1,28 @@
+{ lib
+, fetchFromGitHub
+, rustPlatform
+, darwin
```
Comment (momeemt) [inline pkgs/by-name/du/dummyhttp/package.nix:4]:
```suggestion
, darwin
, stdenv
```

It may be necessary for reasons discussed below.

**Review by momeemt — CHANGES_REQUESTED** [review]
Thank you for your contribution. I left the comments about to build it in *-darwin and minor details. I would be grateful if you could confirm this.

File: pkgs/by-name/du/dummyhttp/package.nix (hunk 1 above)
Comment (GuillaumeDesforges) [reply pkgs/by-name/du/dummyhttp/package.nix:24]: TBH I'd rather not, but I don't want this to be blocking. What's the policy?

**Comment by GuillaumeDesforges** [comment]
Thanks for the review, I'll apply suggestions asap.

File: pkgs/by-name/du/dummyhttp/package.nix (hunk 1 above)
Comment (momeemt) [reply pkgs/by-name/du/dummyhttp/package.nix:24]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes

> `meta.maintainers` must be set for new packages.

Nixpkgs requests to add maintainer name to `meta.maintainers`.

**Review by momeemt — APPROVED** [review]
LGTM, but I don't have enough reviewing experience, so I'm waiting for committers or experienced users to judge it.

**Comment by DanNixon** [comment]
Result of `nixpkgs-review pr 308567` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dummyhttp</li>
  </ul>
</details>

**Review by DanNixon — APPROVED** [review]

File: pkgs/by-name/du/dummyhttp/package.nix [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  cargoHash = "sha256-JkA0qW/MQH+XmiD9eiT0s70HxNNYyk9ecBo4k5nUF10=";
+
+  buildInputs = lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk.frameworks.Security
+  ];
+
+  meta = with lib; {
+    description = "Super simple HTTP server that replies a fixed body with a fixed response code";
+    homepage = "https://github.com/svenstaro/dummyhttp";
+    license = with licenses; [ mit ];
```
Comment (Aleksanaa) [inline pkgs/by-name/du/dummyhttp/package.nix:28]:
```suggestion
    license = licenses.mit;
```

**Review by Aleksanaa — COMMENTED** [review]
Others lgtm

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**